frontend/apps/desktop/src/app/login-overlay/login-overlay.component.ts

```diff
@@ -1,26 +1,69 @@
 import { CommonModule } from '@angular/common';
 import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { ReactiveFormsModule } from '@angular/forms';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { AuthService } from '../business-rules/auth/auth.service';
 
 import { loginFormControl } from '../../../../../libs/business-domain/login/src/lib/form-controls/login-form.control';
-import { ButtonColorEnum, ButtonComponent, ButtonTypeEnum } from '../ui-shared';
+import { isEmail } from '../../../../../libs/business-domain/login/src/lib/rules/is-email.rule';
+import { ButtonColorEnum, ButtonComponent, ButtonFunctionType,ButtonTypeEnum } from '../ui-shared';
+import { TranslatePipe } from '@ngx-translate/core';
 
 @Component({
   selector: 'login-overlay',
   standalone: true,
-  imports: [CommonModule, ReactiveFormsModule, ButtonComponent],
+  imports: [
+    CommonModule,
+    ReactiveFormsModule,
+    ButtonComponent,
+    TranslatePipe
+  ],
   templateUrl: './login-overlay.component.html',
   styleUrl: './login-overlay.component.less',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class LoginOverlayComponent {
   protected readonly ButtonColorEnum = ButtonColorEnum;
   protected readonly ButtonTypeEnum = ButtonTypeEnum;
+  protected readonly ButtonFunctionType = ButtonFunctionType;
   protected readonly form = loginFormControl;
 
+  constructor(private authService: AuthService, private router: Router) {}
+
   public onSubmit(): void {
-    if (this.form.valid) {
-      console.log('Form is valid');
+    if (!this.form.valid) {
+      this.markAllFieldsAsTouched(this.form);
+      return;
+    }
+
+    if (this.form.controls.password.value) {
```

Comment:
Hier würde ich vorher so etwas wie 
`if (!this.form.controls.password.value) {
      return;
  }`
  schöner finden, sonst würde, wenn der Benutzer (noch) kein Passwort eingegeben hat, dieser zu tournament-overview weitergeleitet werden.
 Dann kannst du auch die Verschachtelung reduzieren.